test/test_with_xpress.jl

```diff
@@ -1532,3 +1532,96 @@ end
     end
 end
 
+@testset "Custom REopt logger" begin
+    
+    # Throw a handled error
+    d = JSON.parsefile("./scenarios/logger.json")
+
+    m1 = Model(Xpress.Optimizer)
+    m2 = Model(Xpress.Optimizer)
+    r = run_reopt([m1,m2], d)
+    @test r["status"] == "error"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+    @test length(r["Messages"]["errors"]) > 0
+    @test length(r["Messages"]["warnings"]) > 0
+
+    m = Model(Xpress.Optimizer)
+    r = run_reopt(m, d)
+    @test r["status"] == "error"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+    @test length(r["Messages"]["errors"]) > 0
+    @test length(r["Messages"]["warnings"]) > 0
+
+    # Type is dict when errors, otherwise type REoptInputs
+    @test isa(REoptInputs(d), Dict)
+
+    # Using filepath
+    n1 = Model(Xpress.Optimizer)
+    n2 = Model(Xpress.Optimizer)
+    r = run_reopt([n1,n2], "./scenarios/logger.json")
+    @test r["status"] == "error"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+    @test length(r["Messages"]["errors"]) > 0
+    @test length(r["Messages"]["warnings"]) > 0
+
+    n = Model(Xpress.Optimizer)
+    r = run_reopt(n, "./scenarios/logger.json")
+    @test r["status"] == "error"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+    @test length(r["Messages"]["errors"]) > 0
+    @test length(r["Messages"]["warnings"]) > 0
+
+    # Throw an unhandled error: Bad URDB rate -> stack gets returned for debugging
```

Comment:
I'm curious if it's really necessary to test all 4 run_reopt() versions (1 and 2 models, file path and dict) here again if you already did that above?

Reply:
I think its okay to have these tests. Mainly the first 4 tests are looking at invoking REopt.jl in 4 different ways and handling an error which REopt.jl already handles with `throw(@error(...))` statement. The last 4 tests do the same thing as first, but the error generated now is an unhandled error where REopt is supposed to return the stack + error message. So these tests just try different ways to test that both iterations of `handle_errors()` function are working as expected.

I didn't see any considerable impact on solve times since the scenarios are supposed to error out before they begin solving.